wallet/wallet.go

```diff
@@ -2382,6 +2382,88 @@ func (w *Wallet) TransactionSummary(txHash *chainhash.Hash) (*TransactionSummary
 	return txSummary, err
 }
 
+// GetTicketsResult response struct for gettickets rpc request
+type GetTicketsResult struct {
+	Tickets []*TicketSummary
+}
+
+// GetTickets implements the rpc request command for gettickets
+func (w *Wallet) GetTickets(startBlock, endBlock *BlockIdentifier, cancel <-chan struct{}) (*GetTicketsResult, error) {
+	var start, end int32 = 0, -1
+
+	if startBlock != nil {
+		if startBlock.hash == nil {
+			start = startBlock.height
+		} else {
+			err := walletdb.View(w.db, func(dbtx walletdb.ReadTx) error {
+				ns := dbtx.ReadBucket(wtxmgrNamespaceKey)
+				meta, err := w.TxStore.GetBlockMetaForHash(ns, startBlock.hash)
+				if err != nil {
+					return err
+				}
+				start = meta.Height
+				return nil
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	if endBlock != nil {
+		if endBlock.hash == nil {
+			end = endBlock.height
+		} else {
+			err := walletdb.View(w.db, func(dbtx walletdb.ReadTx) error {
+				ns := dbtx.ReadBucket(wtxmgrNamespaceKey)
+				meta, err := w.TxStore.GetBlockMetaForHash(ns, endBlock.hash)
+				if err != nil {
+					return err
+				}
+				end = meta.Height
+				return nil
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	var res GetTicketsResult
+	err := walletdb.View(w.db, func(dbtx walletdb.ReadTx) error {
+		txmgrNs := dbtx.ReadBucket(wtxmgrNamespaceKey)
+
+		rangeFn := func(details []udb.TxDetails) (bool, error) {
+			// TODO: probably should make RangeTransactions not reuse the
+			// details backing array memory.
+			dets := make([]udb.TxDetails, len(details))
+			copy(dets, details)
+			details = dets
+			for i := range details {
+				// XXX Here is where I would look up the ticket information from the db so I can populate spenderhash and ticket status
+				ticketInfo, err := w.TxStore.TicketDetails(txmgrNs, &details[i])
+				if err != nil {
+					log.Errorf("error while trying to get ticket details for ")
```

Comment:
this should return the error instead of logging (and the log is missing the details about what was looked up too)